src/axom/primal/operators/winding_number.hpp

```diff
@@ -444,6 +458,275 @@ int winding_number(const Point<T, 3>& query,
   return std::lround(wn);
 }
 
+/*!
+ * \brief Computes the solid angle winding number for a Bezier patch
+ *
+ * \param [in] query The query point to test
+ * \param [in] bPatch The Bezier patch object
+ * \param [in] edge_tol The physical distance level at which objects are 
+ *                      considered indistinguishable
+ * \param [in] quad_tol The maximum relative error allowed in the quadrature
+ * \param [in] EPS Miscellaneous numerical tolerance level for nonphysical distances
+ * 
+ * Computes the generalized winding number for a Bezier patch using Stokes theorem.
+ *
+ * \return double The generalized winding number.
+ */
+template <typename T>
+double winding_number(const Point<T, 3>& query,
+                      const BezierPatch<T>& bPatch,
+                      const double edge_tol = 1e-8,
+                      const double quad_tol = 1e-8,
+                      const double EPS = 1e-8)
+{
+  const int ord_u = bPatch.getOrder_u();
+  const int ord_v = bPatch.getOrder_v();
+  const bool patchIsRational = bPatch.isRational();
+  const double edge_tol_sq = edge_tol * edge_tol;
+
+  // Fix the number of quadrature nodes arbitrarily, but high enough
+  //  to `catch` near singularities for refinement
+  const int quad_npts = 50;
+
+  // Early return if the patch is approximately polygonal.
+  //  Very slight variations in curvature requires small EPS tolerance
+  if(bPatch.isPolygonal(EPS))
+  {
+    return winding_number(
+      query,
+      Polygon<T, 3>(axom::Array<Point<T, 3>>(
+        {bPatch(0, 0), bPatch(ord_u, 0), bPatch(ord_u, ord_v), bPatch(0, ord_v)})),
+      edge_tol,
+      PRIMAL_TINY);
+  }
+
+  // Use a specific kind of recursion if we are within tol of an endpoint.
+  //  Split the surface closer to the corner, assume smallest patch is polygonal,
+  //  and set a new edge_tol so corners of the new patch aren't marked as coincident
+  if(squared_distance(query, bPatch(0, 0)) <= edge_tol_sq)
+  {
+    BezierPatch<T> p1, p2, p3, p4;
+    bPatch.split(0.01, 0.01, p1, p2, p3, p4);
```

Comment:
From a readability standpoint, yes I think it would make sense to make it a variable, but I don't think it should be a parameter, or otherwise user accessible. The offset needs to be small, or else the result is incorrect. 

On a similar note, this corner case relies on the idea that after this split is performed, the patch containing the coincident corner is small and polygonal enough that it returns a winding number of zero. If the surface parameterization is really adversarial, this isn't going to be the case. It would be possible to keep checking values of this `edge_offset` value until that's the case, but that would incur a not-insignificant cost for the non-adversarial cases. What do you think?

Reply:
Yes, I meant as a variable in the function for readability, not a function parameter.

Reply:
Let's hold off on handing the above adversarial case until/unless it seems necessary. 
Perhaps make a note about it in the code here though to help us remember.